autogluon_zeroshot/contexts/context.py

```diff
@@ -0,0 +1,329 @@
+import time
+from typing import Dict, List
+from pathlib import Path
+
+import boto3
+from botocore.errorfactory import ClientError
+import pandas as pd
+
+from autogluon.common.loaders import load_pd
+from autogluon.common.utils.s3_utils import download_s3_files
+from autogluon.common.utils.s3_utils import is_s3_url, s3_path_to_bucket_prefix
+
+from .utils import load_zeroshot_input
+from ..loaders import load_configs, load_results, combine_results_with_score_val, Paths
+from ..simulation.simulation_context import ZeroshotSimulatorContext
+
+
+# TODO: Add logic to download files if they don't exist
+class BenchmarkPaths:
+    def __init__(self,
+                 result: str,
+                 results_by_dataset: str,
+                 raw: str,
+                 comparison: str,
+                 task_metadata: str = None,
+                 zs_pp: str = None,
+                 zs_gt: str = None,
+                 configs: List[str] = None,
+                 ):
+        self.result = result
+        self.results_by_dataset = results_by_dataset
+        self.raw = raw
+        self.comparison = comparison
+        self.task_metadata = task_metadata
+        self.zs_pp = zs_pp
+        self.zs_gt = zs_gt
+
+        if configs is None:
+            configs_prefix_1 = str(Paths.data_root / 'configs/configs_20221004')
+            configs_prefix_2 = str(Paths.data_root / 'configs')
+            configs = [
+                f'{configs_prefix_1}/configs_catboost.json',
+                f'{configs_prefix_1}/configs_fastai.json',
+                f'{configs_prefix_1}/configs_lightgbm.json',
+                f'{configs_prefix_1}/configs_nn_torch.json',
+                f'{configs_prefix_1}/configs_xgboost.json',
+                f'{configs_prefix_2}/configs_rf.json',
+                f'{configs_prefix_2}/configs_xt.json',
+                f'{configs_prefix_2}/configs_knn.json',
+            ]
+        self.configs = configs
+
+    def print_summary(self):
+        print(f'BenchmarkPaths Summary:\n'
+              f'\tresult             = {self.result}\n'
+              f'\tresults_by_dataset = {self.results_by_dataset}\n'
+              f'\traw                = {self.raw}\n'
+              f'\tcomparison         = {self.comparison}\n'
+              f'\ttask_metadata      = {self.task_metadata}\n'
+              f'\tzs_pp              = {self.zs_pp}\n'
+              f'\tzs_gt              = {self.zs_gt}\n'
+              f'\tconfigs            = {self.configs}')
```

Comment:
This is great! However I think the summary I have currently is nicer to read:

print(self)
```
BenchmarkPaths(result='/home/ubuntu/workspace/code/autogluon-zeroshot/data/results/all_v3/results_ranked_valid.csv', results_by_dataset='/home/ubuntu/workspace/code/autogluon-zeroshot/data/results/all_v3/results_ranked_by_dataset_valid.parquet', raw='/home/ubuntu/workspace/code/autogluon-zeroshot/data/results/all_v3/openml_ag_2022_10_13_zs_models.parquet', comparison='/home/ubuntu/workspace/code/autogluon-zeroshot/data/results/automl/results_ranked_by_dataset_valid.csv', task_metadata='/home/ubuntu/workspace/code/autogluon-zeroshot/data/metadata/task_metadata.csv', zs_pp='/home/ubuntu/workspace/code/autogluon-zeroshot/data/results/all_v3/zeroshot_pred_proba_2022_10_13_zs.pkl', zs_gt='/home/ubuntu/workspace/code/autogluon-zeroshot/data/results/all_v3/zeroshot_gt_2022_10_13_zs.pkl', configs=['/home/ubuntu/workspace/code/autogluon-zeroshot/data/configs/configs_20221004/configs_catboost.json', '/home/ubuntu/workspace/code/autogluon-zeroshot/data/configs/configs_20221004/configs_fastai.json', '/home/ubuntu/workspace/code/autogluon-zeroshot/data/configs/configs_20221004/configs_lightgbm.json', '/home/ubuntu/workspace/code/autogluon-zeroshot/data/configs/configs_20221004/configs_nn_torch.json', '/home/ubuntu/workspace/code/autogluon-zeroshot/data/configs/configs_20221004/configs_xgboost.json', '/home/ubuntu/workspace/code/autogluon-zeroshot/data/configs/configs_rf.json', '/home/ubuntu/workspace/code/autogluon-zeroshot/data/configs/configs_xt.json', '/home/ubuntu/workspace/code/autogluon-zeroshot/data/configs/configs_knn.json'])
```

print_summary()
```
BenchmarkPaths Summary:
	result             = /home/ubuntu/workspace/code/autogluon-zeroshot/data/results/all_v3/results_ranked_valid.csv
	results_by_dataset = /home/ubuntu/workspace/code/autogluon-zeroshot/data/results/all_v3/results_ranked_by_dataset_valid.parquet
	raw                = /home/ubuntu/workspace/code/autogluon-zeroshot/data/results/all_v3/openml_ag_2022_10_13_zs_models.parquet
	comparison         = /home/ubuntu/workspace/code/autogluon-zeroshot/data/results/automl/results_ranked_by_dataset_valid.csv
	task_metadata      = /home/ubuntu/workspace/code/autogluon-zeroshot/data/metadata/task_metadata.csv
	zs_pp              = /home/ubuntu/workspace/code/autogluon-zeroshot/data/results/all_v3/zeroshot_pred_proba_2022_10_13_zs.pkl
	zs_gt              = /home/ubuntu/workspace/code/autogluon-zeroshot/data/results/all_v3/zeroshot_gt_2022_10_13_zs.pkl
	configs            = ['/home/ubuntu/workspace/code/autogluon-zeroshot/data/configs/configs_20221004/configs_catboost.json', '/home/ubuntu/workspace/code/autogluon-zeroshot/data/configs/configs_20221004/configs_fastai.json', '/home/ubuntu/workspace/code/autogluon-zeroshot/data/configs/configs_20221004/configs_lightgbm.json', '/home/ubuntu/workspace/code/autogluon-zeroshot/data/configs/configs_20221004/configs_nn_torch.json', '/home/ubuntu/workspace/code/autogluon-zeroshot/data/configs/configs_20221004/configs_xgboost.json', '/home/ubuntu/workspace/code/autogluon-zeroshot/data/configs/configs_rf.json', '/home/ubuntu/workspace/code/autogluon-zeroshot/data/configs/configs_xt.json', '/home/ubuntu/workspace/code/autogluon-zeroshot/data/configs/configs_knn.json']

```

Reply:
Ok I get your point with the nice formatting. 

We can have the same in one line to avoid the need to add manually each field, for instance by doing this:

```python
print(f'BenchmarkPaths Summary:\n')
print("".join(f'\t{key} = {value}\n' for key, value in self.__dict__.items()))
 ```

Reply:
Nice! I've edited your code slightly to also have the nice uniform spacing, this is much better than what I had before, thanks!